apicapi/apic_manager.py

```diff
@@ -955,30 +958,46 @@ def ensure_static_endpoint_deleted(self, tenant_id, epg_id, mac_address,
     def add_staticlinks(self):
         # add static hostlinks in config
         for switch in self.switch_dict:
+            pod_id = self.switch_dict[switch].get('pod_id', '1')
             for module_port in self.switch_dict[switch]:
+                if module_port == 'pod_id':
+                    continue
                 module, port = module_port.split('/', 1)
                 hosts = self.switch_dict[switch][module_port]
                 for host in hosts:
-                    self.add_hostlink(host, 'static', None, switch, module,
-                                      port)
+                    try:
+                        host, interface = host.split('|', 1)
+                    except ValueError:
+                        interface = 'static'
+                    self.add_hostlink(host, interface, None, switch, module,
+                                      port, pod_id=pod_id)
 
     def add_hostlink(self, host, ifname, ifmac, switch, module, port,
-                     transaction=None):
+                     pod_id, transaction=None):
         if switch in self.vpc_dict:
             self.add_vpclink(host, ifname, ifmac, switch, module, port,
-                             transaction=None)
+                             pod_id, transaction=None)
             return
 
         # detect old link (say, if changing port on switch)
         hostlinks = []
         for hlink in self.db.get_switch_and_port_for_host(host):
             if hlink[0] == switch:
-                if hlink == (switch, module, port):
-                    # add is no-op, it already exists in DB
-                    return
-                else:
-                    # any other link to the same switch is old
-                    hostlinks.append(hlink)
+                try:
+                    if hlink == (switch, module, port):
```

Comment:
hlink[0:3] == (switch, module, port)

Reply:
Here I need to distinguish if hlink is 3 elements (for legacy plugin) or 5 elements (for AIM) so I think we still need this try and catch block. So I will leave it as is then. Or maybe I miss your point here?